File: NoteBook/src/main/java/NoteBook.java
import java.util.ArrayList;

/**
 * This class is notebook, that contains of notes.
 */

public class NoteBook {
  private ArrayList<Note> notes;
  public NoteBook(ArrayList<Note> notes) {
    this.notes = notes;
  }
  public ArrayList<Note> getNotes() {
    return this.notes;
  }
  public void setNotes(ArrayList<Note> notes) {
    this.notes = notes;
  }
  /**
   * This method is override hashCode method for notebook class.
   * @return - hashCode of notebook.
   */
  @Override
  public int hashCode() {
    final int prime = 31;
    int result = 1;
    result = prime * result + notes.hashCode();
    return result;
  }

  /**
   * This method override equals method for notebook class.
   * @param obj - notebook, that is compared.
   * @return - true, if notebooks are equals, false - in another case.
   */
  @Override
  public boolean equals(Object obj) {
    if (obj == this) return true;
    if (!(obj instanceof Note)) {
      return false;
    }
    Note note = (Note) obj;
    return notes.equals(notes);
  }


  /**
   * This method override toString method for notebook class.
   * @return - one notebook in string presentation.
   */
  @Override
  public String toString() {
    String notesOnTheScreen = "";
    for (int i = 0; i < notes.size(); i++) {
      notesOnTheScreen += notes.get(i).toString();
    }
    return notesOnTheScreen;
  }
}

File: Tasks/Notepad/Notepad/NoteBook.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Notepad
{
  /// <summary>
  /// Class describes essense of notebook
  /// with list of notes.
  /// Contains unique id and
  /// date time of creating
  /// </summary>
  public class NoteBook
  {
    private static int id;

    public string Name { get; set; }
    public List<Note> Notes { get; set; }
    public DateTime Date { get; }
    public int Id { get; }

    /// <summary>
    /// Set field Name like input param
    /// Set current time in field date
    /// Set to notebook id
    /// </summary>
    /// <param name="name">new notebooks name</param>
    public NoteBook(string name)
    {
      Name = name;
      Notes = new List<Note>();
      Date = DateTime.Now;
      Id = id++;
    }

    /// <summary>
    /// Overrided equals method
    /// Check name and list notes fields in notebooks
    /// </summary>
    /// <param name="obj">checked on equality node</param>
    /// <returns>true all fields are equals, false otherwise</returns>
    public override bool Equals(object obj)
    {
      if (obj == null || GetType() != obj.GetType())
      {
        return false;
      }

      NoteBook note = (NoteBook)obj;
      return (Name == note.Name) && (Notes == note.Notes);
    }

    /// <summary>
    /// Overrided get hashcode method
    /// Hashcode is date of creating in milliseconds
    /// </summary>
    /// <returns>Integer hashcode</returns>
    public override int GetHashCode()
    {
      return Date.Millisecond;
    }

    /// <summary>
    /// Overrided toString method
    /// Looks like id+title+date+content
    /// </summary>
    /// <returns>Integer hashcode</returns>
    public override string ToString()
    {
      StringBuilder sb = new StringBuilder(Id).AppendLine();
      return sb.ToString();
    }
  }
}
